### PR TITLE
Adds migration guides to some of the most common legacy functions

### PR DESCRIPTION
**Pull request by qxjit**
This begins adding migration guides for moving from legacy orville to
the new orville. More migration guides will be coming as well.

### CODE REVIEW

File: orville-postgresql-legacy/src/Database/Orville/PostgreSQL/Core.hs [hunk 1]
```diff
@@ -2,11 +2,15 @@
 Module    : Database.Orville.PostgreSQL.Core
 Copyright : Flipstone Technology Partners 2016-2018
 License   : MIT
+
+Migration Guide: Although not all exports are identical, most of the items in
+this module can now be import from @Orville.PostgreSQL@.
```
Comment (AugmenTab) [inline orville-postgresql-legacy/src/Database/Orville/PostgreSQL/Core.hs:7]: import -> imported

File: orville-postgresql-legacy/src/Database/Orville/PostgreSQL/Core.hs [hunk 2]
```diff
@@ -298,6 +326,15 @@ findRecord tableDef key =
   let keyDef = tablePrimaryKey tableDef
    in selectFirst tableDef (where_ $ primaryKeyEquals keyDef key)
 
+{- |
+  Migration Guide: @updateFields@ has been renamed to
+  @updateFieldsAndReturnRowCount@, but now takes a @NonEmpty SetClause@ instead
+  of a @[Field Update]@ and a @Maybe BooleanExpr@ instead of a
+  @[WhereCondition]@.
+
+  @updateFields@ still exists a variant of this function, but returns @()@
```
Comment (AugmenTab) [inline orville-postgresql-legacy/src/Database/Orville/PostgreSQL/Core.hs:335]: Would this be "exists as a variant?"

File: orville-postgresql-legacy/src/Database/Orville/PostgreSQL/Core.hs [hunk 3]
```diff
@@ -327,6 +369,11 @@ updateRecord tableDef key record = do
       updates = zipWith FieldUpdate fields (runToSql builder record)
   void $ updateFields tableDef updates conds
 
+{- |
+  Migration Guide: @insertRecord@ has been renamed to @insertAndReturnEntity@.
+  Note there are is are also new variant functions @insertEntity@ and
```
Comment (AugmenTab) [inline orville-postgresql-legacy/src/Database/Orville/PostgreSQL/Core.hs:374]: ~~is are~~

File: orville-postgresql-legacy/src/Database/Orville/PostgreSQL/Core.hs [hunk 4]
```diff
@@ -383,6 +440,12 @@ insertRecordMany tableDef newRecords = do
         insert <- prepare conn insertSql
         void $ execute insert (concatMap (runToSql builder) newRecords)
 
+{- |
+  Migration Guide: @deleteRecord@ has been renamed to @deleteEntity@. Note
+  that there are also new variant function @deleteAndReturnEntity@ and
```
Comment (AugmenTab) [inline orville-postgresql-legacy/src/Database/Orville/PostgreSQL/Core.hs:445]: "function" should be pluralized here.

File: orville-postgresql-legacy/src/Database/Orville/PostgreSQL/Internal/RelationalMap.hs [hunk 5]
```diff
@@ -61,6 +64,13 @@ data TableParams readEntity writeEntity key = TableParams
   }
 
 {-|
+  Migration Guide: This function has in the new orville to take the table name,
+  primary key definition and a @SqlMarshaller@ (formerly @RelationalMap@).
+  Other options such as constraints, indexes, and columns to drop can be added
+  to the @TableDefinition@ after the initial instatiation. The @TableParams@
```
Comment (AugmenTab) [inline orville-postgresql-legacy/src/Database/Orville/PostgreSQL/Internal/RelationalMap.hs:70]: instatiation -> instantiation

File: orville-postgresql-legacy/src/Database/Orville/PostgreSQL/Internal/Types.hs [hunk 6]
```diff
@@ -296,6 +296,13 @@ getComponent getComp (ToSql serializer) =
   ToSql (withReaderT getComp serializer)
 
 {-|
+ Migration Guide: 'TableDefinition' can now be imported from
+ 'Orville.PostgreSQL'. The ordered of the type parameter has changed from
```
Comment (AugmenTab) [inline orville-postgresql-legacy/src/Database/Orville/PostgreSQL/Internal/Types.hs:300]: ordered -> order

File: orville-postgresql-legacy/src/Database/Orville/PostgreSQL/Internal/Types.hs [hunk 7]
```diff
@@ -296,6 +296,13 @@ getComponent getComp (ToSql serializer) =
...
+ @TableDefinition readEnity writeEntity key@ to @TableDefinition key
+ writeEntity readEntity@. In the new Orville tables without primary keys are
+ supported, so the @key@ parameter must now be instatiated as either @HasKey
```
Comment (AugmenTab) [inline orville-postgresql-legacy/src/Database/Orville/PostgreSQL/Internal/Types.hs:303]: instatiated -> instantiated

**Review by AugmenTab — APPROVED** [review]

File: orville-postgresql-legacy/src/Database/Orville/PostgreSQL/Core.hs (hunk 2 above)
Comment (qxjit) [reply orville-postgresql-legacy/src/Database/Orville/PostgreSQL/Core.hs:335]: Yes! :heart:

**Pull request merged**